### DISCUSSION TITLE
[Adonis v5] How to iterate trough related?

### DISCUSSION THREAD

**Discussion by melokki** [adonisjs/core#1716] [Help] [answered]
Hello again :wave: :upside_down_face: 

I am now in a Lucid model and I am making a computed property. For this I need to iterate over a relationship 
```js
// this code successfully loads my relationship, but I cannot iterate over the array
const payments = this.$getRelated("payments");
```
this is the error I am getting
```
Property 'forEach' does not exist on type 'OneOrMany<LucidRow>'.
  Property 'forEach' does not exist on type 'LucidRow'.
```
I tried to use other array methods too, but I got the same error

Am I doing something wrong or this is just not possible?

**Comment by thetutlage** [comment, marked as answer]
Why not simply use `this.payments`?

**Reply by melokki** [reply]
Thank you for your reply, yup, that's what I ended up using
I was thinking that if I use `$getRelated()` it will auto preload the relationship for me (is that even possible? :thinking: )

**Reply by thetutlage** [reply]
Nope. Preloading has to be an explicit function call. No implicit magic is done in AdonisJS codebase 🙂

**Reply by melokki** [reply]
:upside_down_face:  Understood 
thank you very much
